### ISSUE TITLE
Cop that replaces + with \ for line continuation (for strings) does not let quoting cop run

### ISSUE THREAD

**Issue by stamhankar999** [bbatsov/rubocop#2738]
After replacing + with \ for putting two strings together, it should replace double-quote with single-quote when appropriate.

Example:
> "foo bar" + 
>   "#{baz}"```

should become
> 'foo bar' \
>    "#{baz}"

but the foo bar line maintains it's double-quote unnecessarily[

**Comment by jonas054** [comment]
If you have `Style/StringLiterals` enabled and configured with
```yaml
Style/StringLiterals:
  EnforcedStyle: single_quotes
```
it will auto-correct
```ruby
"foo bar" \
"#{baz}"
```
into
```ruby
'foo bar' \
"#{baz}"
```
